core/domain/draft_upgrade_services.py

```diff
@@ -88,6 +95,122 @@ def try_upgrading_draft_to_exp_version(
 class DraftUpgradeUtil(python_utils.OBJECT):
     """Wrapper class that contains util functions to upgrade drafts."""
 
+    @classmethod
+    def _convert_states_v32_dict_to_v33_dict(cls, exp_id, draft_change_list):
+        """Converts draft change list from state version 32 to 33. State
+        version 33 adds image dimensions to images inside collapsible
+        blocks and tabs, for which there should be no changes to drafts.
+
+        Args:
+            exp_id: str. Exploration id.
+            draft_change_list: list(ExplorationChange). The list of
+                ExplorationChange domain objects to upgrade.
+
+        Returns:
+            list(ExplorationChange). The converted draft_change_list.
+        """
+        file_system_class = fs_services.get_entity_file_system_class()
+        file_system = fs_domain.AbstractFileSystem(file_system_class(
+            feconf.ENTITY_TYPE_EXPLORATION, exp_id))
+        for i, change in enumerate(draft_change_list):
+            # Changes for html in state content.
+            if (change.cmd == exp_domain.CMD_EDIT_STATE_PROPERTY and
+                    change.property_name == exp_domain.STATE_PROPERTY_CONTENT):
+                html_string = change.new_value['html']
+                converted_html_string = (
+                    html_validation_service.add_dimensions_to_image_tags_inside_tabs_and_collapsible_blocks( # pylint: disable=line-too-long
+                        file_system, html_string))
+                draft_change_list[i] = exp_domain.ExplorationChange({
```

Comment:
Here and elsewhere ... can't you just update draft_change_list[i].new_value.html directly?